synapse/storage/persist_events.py

```diff
@@ -131,22 +169,26 @@ def add_to_queue(self, room_id, events_and_contexts, backfilled):
 
         deferred = ObservableDeferred(defer.Deferred(), consumeErrors=True)
 
+        span = opentracing.active_span()
         queue.append(
-            self._EventPersistQueueItem(
+            _EventPersistQueueItem(
                 events_and_contexts=events_and_contexts,
                 backfilled=backfilled,
                 deferred=deferred,
+                opentracing_context=span.context if span else None,
```

Comment:
I'm a bit concerned that this is going to cause confusion, given that we'll only be linking together the persist span with the *first* request that added events into this batch (rather than all of them). It'll likely work 99% of the time as usually there will only be events from one request in the queue at any given time.

Ideally we'd link the persist span to all the request spans, which can be done using `follows_from` refs, but a) I have a suspicion that the debug sampling does the right thing there (in that iirc it only looks at the first parent?), and b) the UI doesn't really support `follows_from` with mutliple parents :/

Reply:
oh. yes. I'd forgotten that we could extend batches.

And yes, I spent a while experimenting with multiple parents/follows_from. The UI only seems to let you go from the child to the parent, not the parent to the child. Which might be enough, I suppose, if we create a `event_persist_result` span for each request...